lib/upload.js

```diff
@@ -61,52 +67,36 @@ module.exports = {
         params.push('message=' + encodeURIComponent(message));
       }
       var url = api.uploadUrl + '/upload/' + firebase + '?' + params.join('&');
-      var readStream = fs.createReadStream(filename);
 
       var r = request.put({
         url: url,
         json: true
       }, function(err, response, body) {
-        fs.unlink(filename);
         var failed = (err || !body || !body.success);
         setTimeout(callback, 0, failed, body ? body.directory : undefined);
       });
       var form = r.form();
-      form.append('site', readStream);
+      form.append('site', bufferStream, {
+        filename: 'site.tar.gz',
+        contentType: 'application/x-gzip',
+        knownLength: tarballSize
+      });
     });
   },
   deleteSite: function(firebase, pushId, message, callback) {
-    var writeStream = temp.createWriteStream({ suffix: '.tar.gz' }),
-        filename = writeStream.path,
-        publicDir = temp.mkdirSync();
+    var params = ['id=' + encodeURIComponent(pushId), 'token=' + auth.token];
+    if (message) {
+      params.push('message=' + encodeURIComponent(message));
+    }
+    var url = api.uploadUrl + '/upload/' + firebase + '?' + params.join('&');
 
-    console.log('Preparing to delete site...');
-
-    fstreamIgnore({
-      path: publicDir,
-      type: 'Directory'
-    }).pipe(tar.Pack())
-      .pipe(zlib.createGzip())
-      .pipe(writeStream);
-
-    writeStream.once('finish', function() {
-      var params = ['id=' + encodeURIComponent(pushId), 'fileCount=0', 'token=' + auth.token];
-      if (message) {
-        params.push('message=' + encodeURIComponent(message));
-      }
-      var url = api.uploadUrl + '/upload/' + firebase + '?' + params.join('&');
-      var readStream = fs.createReadStream(filename);
-
-      var r = request.put({
-        url: url,
-        json: true
-      }, function(err, response, body) {
-        fs.unlink(filename);
-        var failed = (err || !body || !body.success);
-        setTimeout(callback, 0, failed, body ? body.directory : undefined);
-      });
-      var form = r.form();
-      form.append('site', readStream);
+    var r = request({
+      url: url,
+      method: 'DELETE',
+      json: true
+    }, function(err, response, body) {
+      var failed = (err || !body || !body.success);
+      setTimeout(callback, 0, failed, body ? body.directory : undefined);
```

Comment:
`body ? body.directory : undefined` can be simplified to `body && body.directory`